File: sdks/csharp/src/Sudomimus.Token/JwtHeader.cs
using System.Text.Json.Serialization;

namespace Sudomimus.Token;

/// <summary>
/// Exact JOSE protected-header fields shared by Sudomimus application access
/// and refresh tokens. Registered JWT claims live in the payload.
/// </summary>
public sealed record JwtHeader
{
    [JsonPropertyName("alg")]
    public required string Algorithm { get; init; }

    [JsonPropertyName("typ")]
    public required string Type { get; init; }

    [JsonPropertyName("kid")]
    public required string KeyId { get; init; }
}

File: sdks/java/token/src/main/java/com/sudomimus/token/JwtHeader.java
package com.sudomimus.token;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Exact JOSE protected-header fields shared by application access and refresh
 * tokens. Registered JWT claims live in the payload.
 */
public final class JwtHeader {

    @JsonProperty("alg") public String algorithm;
    @JsonProperty("typ") public String type;
    @JsonProperty("kid") public String keyId;
}
